src/client.ts

```diff
@@ -38,15 +63,25 @@ export async function* fetchJsonPaginated<T>(
   }
 }
 
-export function fetchPointerChanges(
+export async function* fetchPointerChanges(
   components: Pick<SnapshotsFetcherComponents, 'fetcher' | 'metrics'>,
   server: string,
-  fromTimestamp: number
-): AsyncIterable<DeploymentWithAuthChain> {
+  fromTimestamp: number,
+  logger: ILoggerComponent.ILogger
+): AsyncIterable<PointerChangesSyncDeployment> {
   const url = new URL(
     `${server}/pointer-changes?sortingOrder=ASC&sortingField=local_timestamp&from=${encodeURIComponent(fromTimestamp)}`
   ).toString()
-  return fetchJsonPaginated(components, url, ($) => $.deltas, 'dcl_catalysts_pointer_changes_response_time_seconds')
+  for await (const deployment of fetchJsonPaginated(components, url, ($) => $.deltas, 'dcl_catalysts_pointer_changes_response_time_seconds')) {
+    if (PointerChangesSyncDeployment.validate(deployment)) {
```

Comment:
Will this work for legacy pointer-changes?
```suggestion
    if (PointerChangesSyncDeployment.validate(deployment)) {
```

Reply:
Hmmmm don't know. Anyways, it's just "moved" logic. That validation was taking place with the type `DeploymentWithAuthChain` which is equivalent to `PointerChangesSyncDeployment`.